### DISCUSSION TITLE
DiD panel data outcome variable

### DISCUSSION THREAD

**Discussion by RShi8** [DoubleML/doubleml-for-py#257] [General]
Hi all,

Can anyone explain the data structure particularly the outcome variable structure that should be fed into DoubleMLData for DML DiD panel data? How is the dataframe requirement for DML DiD differ from classic DiD panel dataset? An example dataframe would be very helpful! 

As shown on https://docs.doubleml.org/stable/examples/py_double_ml_did.html, the difference \delta Y_i should be defined as the outcome in DoubleMLData, but I have difficulty visualizing the dataframe as described. Thanks in advance for the help.

**Comment by SvenKlaassen** [comment]
Hi @RShi8,

Generally the `DoubleMLDID` class only considers two time periods. As such you can directly use an `DoubleMLData` object.
The dataframe looks completely identical to the PLR or IRM models. The only difference is that your outcome is defined as

$$ \Delta y_{i,t} = y_{i, t} - y_{i, t-1}$$

So this is a transformation you have to apply to your data before using DoubleML. 
Maybe the DoubleMLData objects in this extended [DiD Example](https://docs.doubleml.org/stable/examples/py_double_ml_did_pretest.html) are helpful.
